src/python/WMCore/WorkQueue/Policy/Start/MonteCarlo.py

```diff
@@ -45,6 +45,15 @@ def split(self):
         mask = Mask(**self.mask)
 
         #First let's initialize some parameters
+        lumis_per_job = ceil(self.args['SliceSize'] / self.args['SubSliceSize'])
+        totalLumisPerElement = int(self.args['MaxJobsPerElement']) * lumis_per_job
+
+        if self.args['MaxLumisPerElement'] and totalLumisPerElement > int(self.args['MaxLumisPerElement']):
+            # If there are too many lumis in the WQ element. reduce the number of jobs per element
+            self.args['MaxLumisPerElement'] = floor(totalLumisPerElement / lumis_per_job)
```

Comment:
I knew something wasn't right here. After spending some pen+paper time, I think we have to actually replace this L53 by
`self.args['MaxJobsPerElement'] = floor(int(self.args['MaxLumisPerElement']) / lumis_per_job)`

this will give us a smaller number of jobs per element, which is then used to calculate `stepSize` below.

In addition to that, I can't see any use case for this `if` below.

Reply:
Alan, you are completely right! That was I was intend to do. :-)
If statement below is needed, Of course I made another typo. It should be below.
```
if self.args['MaxJobsPerElement'] == 0:

```
I will update the patch and add the unittest. Thanks.